backend/entities/event_entity.py

```diff
@@ -0,0 +1,106 @@
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
+    # Name of the event
+    name: Mapped[str] = mapped_column(String)
+    # Time of the event
+    time: Mapped[datetime] = mapped_column(DateTime)
+    # Location of the event
+    location: Mapped[str] = mapped_column(String)
+    # Description of the event
+    description: Mapped[str] = mapped_column(String)
+    # Whether the event is public or not
+    public: Mapped[bool] = mapped_column(Boolean)
+    
+    # Organization hosting the event
+    # NOTE: This defines a one-to-many relationship between the organization and events tables.
+    organization_id: Mapped[int] = mapped_column(ForeignKey("organization.id"))
+    organization: Mapped['OrganizationEntity'] = relationship(back_populates="events")
+
+
+    @classmethod
+    def from_model(cls, model: Event) -> Self:
+        """
+        Class method that converts an `Event` model into a `EventEntity`
+        
+        Parameters:
+            - model (Event): Model to convert into an entity
+        Returns:
+            EventEntity: Entity created from model
+        """
+        return cls(id=model.id, name=model.name, time=model.time, location=model.location, description=model.description, public=model.public, organization_id=model.organization_id)
+
+    def to_model(self) -> Event:
+        """
+        Converts a `EventEntity` object into a `Event` model object
+        
+        Returns:
+            Event: `Event` object from the entity
+        """
+        return Event(id=self.id, 
+           name=self.name, 
+           time=self.time, 
+           location=self.location, 
+           description=self.description, 
+           public=self.public,
+           organization_id=self.organization_id
+        )
+
+    @classmethod
+    def from_details_model(cls, model: EventDetails):
+        """
+        Class method that converts an `EventDetails` model into a `EventEntity`
+        
+        Parameters:
+            - model (EventDetails): Model to convert into an entity
+        Returns:
+            EventEntity: Entity created from model
+        """
+        return cls(
+            id=model.id, 
+            name=model.name, 
+            time=model.time, 
+            location=model.location, 
+            description=model.description, 
+            public=model.public, 
+            organization_id=model.organization_id
+        )
+
+        
+    def to_details_model(self) -> EventDetails:
+        """Create a EventDetails model from an EventEntity, with permissions and members included.
+
+        Returns:
+            EventDetails: An EventDetails model for API usage.
+        """
+        return EventDetails(
+            id=self.id, 
+            name=self.name, 
+            time=self.time, 
+            location=self.location, 
+            description=self.description, 
+            public=self.public,
+            organization_id=self.organization_id,
```

Comment:
For `EventDetails`, it seems like we should drop the `organization_id` field and let it reside in the nested `organization` object. This will likely touch a few places, but seems like the right way of climbing the abstraction ladder.

Reply:
As mentioned in a comment below, the only way this can work is to have a `NewEvent` model without `organization_id` and then have `EventDetails` inherit from that. Do you think this is what we should move towards?

Reply:
My argument would be to have `Event` not have the relation/id and then let the relation/id reside in `EventDetails#organization.id`. Effectively, anywhere you actually need do anything in the model with the related org, you would have something like:

~~~
{ ...
  organization: {id: 1, ...}
}
~~~

In the modelling, this could be an OrganizationPartial (removing required fields so you only had to worry about 1.

The high-level point being this is an example of where we want the model abstraction layer to hide the detail of the join column identity.

Reply:
@KrisJordan in the create event service function, would there be another way we would want to check for permissions then? We can't necessarily pass EventDetails as the request model because it would lack an Organization, and currently we test for permissions using the `organization_id` passed in through `Events`.

Reply:
(Specifically referencing line 57 in `/services/event.py`